userspace/libsinsp/container.h

```diff
@@ -171,20 +171,41 @@ class sinsp_container_engine_docker
 
 	bool resolve(sinsp_container_manager* manager, sinsp_threadinfo* tinfo, bool query_os_for_missing_info);
 	static void cleanup();
+	static void refresh();
 	static void set_query_image_info(bool query_image_info);
+
+	struct docker_request_info
+	{
+		string url;
+		string buf;
+		Json::Value c_root;
+		Json::Value i_root;
+		bool query_images_endpoint;
+		string id_to_query_for_ip;
+		sinsp_container_info container_info;
+		sinsp_threadinfo *tinfo;
```

Comment:
I think there's a possible race where the thread could go away by the time you've completed the request to the docker daemon and gotten a response back. If so, it's probably safer to provide just the pid and do the lookup of sinsp_threadinfo in parse_docker.